src/inmanta/env.py

```diff
@@ -589,31 +681,29 @@ def get_installed_packages(self, only_editable: bool = False) -> Dict[str, versi
     def install_from_index(
         self,
         requirements: List[Requirement],
-        index_urls: Optional[List[str]] = None,
+        config: PipConfig,
         upgrade: bool = False,
-        allow_pre_releases: bool = False,
         constraint_files: Optional[List[str]] = None,
         upgrade_strategy: PipUpgradeStrategy = PipUpgradeStrategy.ONLY_IF_NEEDED,
-        use_pip_config: Optional[bool] = False,
     ) -> None:
         if len(requirements) == 0:
             raise Exception("install_from_index requires at least one requirement to install")
         constraint_files = constraint_files if constraint_files is not None else []
         inmanta_requirements = self._get_requirements_on_inmanta_package()
-        self._run_pip_install_command(
+
+        Pip.run_pip_install_command_from_config(
             python_path=self.python_path,
+            config=config,
             requirements=[*requirements, *inmanta_requirements],
-            index_urls=index_urls,
+            constraints_files=constraint_files,
             upgrade=upgrade,
-            allow_pre_releases=allow_pre_releases,
-            constraints_files=[*constraint_files],
             upgrade_strategy=upgrade_strategy,
-            use_pip_config=use_pip_config,
         )
 
     def install_from_source(
         self,
         paths: List[LocalPackagePath],
+        pip_config: PipConfig,
```

Comment:
I made this compatible again

Reply:
Looks good. I think that ideally we still want to update that script so that it uses the project's pip config when appropriate (or that it uses whatever we end up with for `inmanta.moduletool`).